db-service/lib/cqn2sql.js

```diff
@@ -252,21 +252,14 @@ class CQN2SQLRenderer {
       return col
     }).flat()
 
-    // Prevent SQLite from hitting function argument limit of 100
-    let obj = ''
+    const isRoot = SELECT.expand === 'root'
 
-    if (cols.length < 50) obj = `json_object(${cols.slice(0, 50)})`
-    else {
-      const chunks = []
-      for (let i = 0; i < cols.length; i += 50) {
-        chunks.push(`json_object(${cols.slice(i, i + 50)})`)
-      }
-      // REVISIT: json_merge is a user defined function, bad performance!
-      obj = `json_merge(${chunks})`
+    // Prevent SQLite from hitting function argument limit of 100
+    let obj = "'{}'"
+    for (let i = 0; i < cols.length; i += 48) {
+      obj = `json${isRoot ? '' : 'b'}_insert(${obj},${cols.slice(i, i + 48)})`
     }
-
-
-    return `SELECT ${SELECT.one || SELECT.expand === 'root' ? obj : `json_group_array(${obj.includes('json_merge') ? `json_insert(${obj})` : obj})`} as _json_ FROM (${sql})`
+    return `SELECT ${isRoot || SELECT.one ? obj : `jsonb_group_array(${obj})`} as _json_ FROM (${sql})`
```

Comment:
just wondering about the sqlite specific stuff in the generic `db-service`, could you elaborate why we need this here?

Reply:
It has always been in the generic `db-service`. The original reason for having it here is that there is no `SQL` specified uniform `API` to create `json` values. Therefor taking `SQLite` makes it easy for other database services to find their equivalent functions.

Reply:
Got it. Thanks!